### PR TITLE
Fix: length_scale_prior silently ignored; raise default to [1e-3, 1e2]

### LINKED ISSUE #3: Bug: length_scale_prior is silently ignored  kernel always uses the default (1e-5, 1e5) length-scale bounds
### Summary

The `length_scale_prior` passed to `GaussianProcessRegressor` never reaches the kernel. It is
handed to `length_corr_kernel` under the **`prior_bounds`** keyword instead of
`length_scale_bounds`, so `length_scale_bounds` silently falls back to its sklearn default
`(1e-5, 1e5)`.

Every GP fit on `main` therefore optimises the length scales over **10 decades** instead of the
intended **4** (`[1e-3, 1e1]`), and `prior_bounds` simultaneously receives an object it was never
meant to hold.

### Reproducer

```python
import numpy as np
from gpry.gpr import GaussianProcessRegressor

lsp = np.array([[1e-3, 1e1]] * 4)          # the documented default prior
g = GaussianProcessRegressor(kernel="RBF", length_scale_prior=lsp)
print(np.exp(g.kernel.bounds)[1:])         # -> [1e-05, 1e+05] per dimension, not [1e-3, 1e1]
print(g.kernel.k2.length_scale_bounds)     # -> (1e-05, 100000.0)  i.e. the untouched default
```

### Root cause

`gpry/gpr.py` (currently ~L201):

```python
) * length_corr_kernel(
    length_scale_init,
    prior_bounds=length_scale_prior,   # <-- wrong keyword
    **kernel_args,
)
```

`RBF.__init__(self, length_scale=1.0, length_scale_bounds=(1e-5, 1e5), prior_bounds=None)` —
`length_scale_bounds` is never passed, so it keeps its default. `prior_bounds` is only consumed
when `length_scale_bounds == "dynamic"` (to compute `max_length`), and `"dynamic"` is not
requested anywhere in the codebase, so the value passed there is inert.

Introduced in **39dc81c** *"split surrogate and gpr: first working (imperfectly) implementation"*
(2025-07-03), the surrogate/GPR split:

```diff
-                ) * length_corr_kernel(
-                    [length_scale_init] * self.d,
-                    length_scale_prior,           # positional -> length_scale_bounds  (correct)
-                    prior_bounds=self.bounds_,    # parameter-space bounds             (correct)
+            ) * length_corr_kernel(
+                length_scale_init,
+                prior_bounds=length_scale_prior,  # wrong slot
```

### Suggested fix

Keep the `length_scale_prior` vocabulary throughout (clearer, and it matches the public API), and
restore the parameter-space bounds so `"dynamic"` works again:

* `length_scale_prior` — the prior on the length-scale *hyperparameter*;
* `prior_bounds` — the *parameter-space* box, in the kernel's (transformed) input space, used to
  derive `max_length` when `length_scale_prior == "dynamic"`.

1. **`gpry/kernels.py`** — rename the constructor argument and attribute
   `length_scale_bounds` → `length_scale_prior` in all four affected kernels, updating the
   `hyperparameter_length_scale` overrides accordingly, and set the default to `(1e-3, 1e1)`.
   This is safe: GPry never calls `super().__init__()` in these kernels and already overrides
   `hyperparameter_length_scale`; sklearn touches `length_scale_bounds` only in those two places,
   and `get_params`/`clone` introspect GPry's own signature. Keep `prior_bounds` as-is.

2. **Guard the four `"dynamic"` comparisons** so an array-valued prior cannot explode:

   ```python
   if isinstance(length_scale_prior, str) and length_scale_prior == "dynamic":
   ```

   This is what forced the original workaround: `length_scale_prior` is now per-dimension, shape
   `(d, 2)`, and `array == "dynamic"` raises.

3. **`gpry/gpr.py`** — give the GPR back the parameter-space bounds as a new optional argument
   (they were lost when `self.bounds_` moved to `SurrogateModel`), and pass both quantities under
   their correct names:

   ```python
   def __init__(self, kernel="RBF", ..., n_restarts_optimizer=0,
                prior_bounds=None,          # NEW: transformed parameter-space bounds
                random_state=None):
   ...
   ) * length_corr_kernel(
       length_scale_init,
       length_scale_prior=length_scale_prior,
       prior_bounds=prior_bounds,
       **kernel_args,
   )
   ```

4. **`gpry/surrogate.py`** — supply it. The required value is already computed in
   `SurrogateModel.__init__` (currently ~L166, for the infinities classifier):

   ```python
   bounds_ = self.preprocessing_X.transform_bounds(self._bounds)
   ...
   kwargs_regressor["prior_bounds"] = bounds_
   ```

   Hoisting that line above the regressor construction is enough; `NormalizeBounds` sets
   `fitted = True` at init, so no ordering problem arises. This reproduces the pre-refactor
   semantics exactly (`prior_bounds = preprocessing_X.transform_bounds(bounds)`), which is the
   transformed space the kernel — and hence `max_length` — lives in.

   The kernels already raise a clear `TypeError` when `"dynamic"` is requested without
   `prior_bounds`, so misuse stays loud.

### PR DESCRIPTION
**Pull request by jonaselgammal**
Fixes #3.

## What was wrong

`length_scale_prior` was passed to `length_corr_kernel` under the **`prior_bounds`** keyword instead of `length_scale_bounds`, so the latter silently kept its sklearn default `(1e-5, 1e5)`. Every fit since 39dc81c (2025-07-03, the surrogate/GPR split) optimised the length scales over **10 decades instead of 4**. In the same refactor `self.bounds_` left the GPR, so the real parameter-space bounds stopped reaching the kernel entirely and `"dynamic"` length-scale bounds became unusable.

## What this PR does

Keeps the `length_scale_prior` vocabulary and keeps `"dynamic"` working.

**`10b1381` — the fix**
- `kernels.py`: rename argument/attribute `length_scale_bounds` → `length_scale_prior` in the four length-correlation kernels (RBF, Matern, RationalQuadratic, ExpSineSquared).
- `kernels.py`: guard the four `"dynamic"` comparisons with `isinstance(..., str)`. This was the blocker: `length_scale_prior` is now per-dimension `(d, 2)`, and `array == "dynamic"` raises — which is presumably why the value was moved to `prior_bounds` in the first place.
- `gpr.py`: new optional `prior_bounds` argument (the parameter-space box **in the GPR's own transformed space**), and both quantities passed under their correct names.
- `surrogate.py`: supplies it as `preprocessing_X.transform_bounds(self._bounds)`, restoring the pre-refactor semantics through the new two-layer architecture.
- `kernels.py`: `length_scale_bounds` property alias, and a `__setstate__` migration so surrogates pickled **before** the rename still load (without it, old checkpoints fail inside sklearn's `get_params`).

**`6ddc057` — default raised to `[1e-3, 1e2]`, plus docs**

With the plumbing fixed, the documented `[1e-3, 1e1]` proved too tight — the optimum lies above it. Measured on real surrogates, clamping length scales to 10 costs a lot of log-marginal-likelihood:

| surrogate | fitted ls median | LML(optimum) | LML(ls=10) | loss |
|---|---|---|---|---|
| d16 nuts | 36.5 | 460.2 | −248.4 | 709 nats |
| d16 ultranest | 37.7 | 578.1 | −318.5 | 897 nats |
| d30 nuts | 47.2 | 875.1 | −1643.3 | 2518 nats |
| d30 ultranest | 45.0 | 417.6 | −2053.1 | 2471 nats |

and at d=4 it pegged 3 of 4 length scales at the upper bound. `[1e-3, 1e2]` also coincides with what `"dynamic"` derives for a unit (normalised) box, so the static default now matches the adaptive behaviour in the common case.

Also corrects two docstrings that contradicted the code:
- `Hyperparameter.dynamic` claimed *"two orders of magnitude above and below the current best fit"*; the code uses `[value * 1e-3, value * 100]`, and only when `max_length` is None — otherwise the bounds scale to the prior size. Both branches are now documented.
- `max_length` claimed it restricts the range to *"2x the prior"*; it is 1e-3× to 100×.

## Verification

A/B over 5 seeds (d=4) against a pristine `origin/main` worktree — recovery unchanged within seed noise, clamping gone:

| | ls median | at upper bound | KL median | KL range | sigma error |
|---|---|---|---|---|---|
| `origin/main` (bug) | 13.50 | 0/20 | 0.00655 | [0.0052, 0.0086] | 0.0144 |
| this PR | 13.29 | **0/20** | 0.00823 | [0.0041, 0.0087] | 0.0124 |

Unit checks, all passing: default prior honoured (`[1e-3, 1e2]`, was `[1e-5, 1e5]`); per-dimension priors honoured; output-scale bounds unchanged; `"dynamic"` derives `[width*1e-3, width*100]` per dimension; `"dynamic"` without `prior_bounds` still raises a clear `TypeError`; pre-rename pickles load; alias works.

`pytest tests/` shows the **same 6 failures as `origin/main`** (pre-existing: stale `test_ranked_pool` API, `mc_sample.to_getdist()` vs dict in `test_pipeline`, MC-return arity and a too-small `max_total` in `test_io`, missing cobaya component). No new failures.

## Notes for review

- This changes the outcome of **every** GP fit, so benchmark numbers produced on `main` since 2025-07-03 should be regenerated.
- `prior_bounds` is still only consumed under `"dynamic"`. It is now plumbed through correctly so the option works, rather than removed.
- Naming: the kernel now takes both `length_scale_prior` (prior on the hyperparameter) and `prior_bounds` (the parameter-space box). Both contain "prior" but mean different things, which is plausibly what made the original mix-up easy; renaming `prior_bounds` → `parameter_bounds` would remove the ambiguity, but I left it out to keep this diff focused.

### CODE REVIEW

File: gpry/kernels.py [hunk 1]
```diff
@@ -76,16 +76,24 @@ class Hyperparameter(
...
-        The prior bounds of the posterior distribution (of the parameter-space,
-        not the hyperparameter space) is required for hyperparameters which are
-        length scales (correlation lengths) if their bounds are set to
-        "dynamic". This is done to restrict their range to the same order of
-        magnitude as the prior size (actually 2x the prior).
+        Size of the parameter-space prior (*not* of the hyperparameter space)
+        per dimension, i.e. ``upper - lower`` of the bounds in the kernel's own
+        (transformed) input space. Only used for length-scale (correlation
+        length) hyperparameters whose bounds are set to "dynamic", to scale
+        their allowed range to the prior size. Derived from the ``prior_bounds``
```
[the remark this answers is not in the document]
Comment (jonaselgammal) [reply gpry/kernels.py:97]: The better way is to document the exceptions
